packages/client/src/generation/TSClient/PrismaClient.ts

```diff
@@ -43,6 +45,28 @@ function clientTypeMapModelsDefinition(this: PrismaClientClass) {
 }`
 }
 
+function clientTypeMapModelsResultDefinition(modelName: string, action: Exclude<Operation, `$${string}`>) {
+  if (action === 'count') return `$Utils.Optional<${getCountAggregateOutputName(modelName)}> | number`
+  if (action === 'groupBy') return `$Utils.Optional<${getGroupByName(modelName)}>[]`
+  if (action === 'aggregate') return `$Utils.Optional<${getAggregateName(modelName)}>`
+  if (action === 'findRaw') return `Prisma.JsonObject`
+  if (action === 'aggregateRaw') return `Prisma.JsonObject`
+  if (action === 'deleteMany') return `Prisma.BatchPayload`
+  if (action === 'createMany') return `Prisma.BatchPayload`
+  if (action === 'updateMany') return `Prisma.BatchPayload`
+  if (action === 'findMany') return `$Utils.PayloadToResult<${modelName}Payload>[]`
+  if (action === 'findFirst') return `$Utils.PayloadToResult<${modelName}Payload> | null`
+  if (action === 'findUnique') return `$Utils.PayloadToResult<${modelName}Payload> | null`
+  if (action === 'findFirstOrThrow') return `$Utils.PayloadToResult<${modelName}Payload>`
+  if (action === 'findUniqueOrThrow') return `$Utils.PayloadToResult<${modelName}Payload>`
+  if (action === 'create') return `$Utils.PayloadToResult<${modelName}Payload>`
+  if (action === 'update') return `$Utils.PayloadToResult<${modelName}Payload>`
+  if (action === 'upsert') return `$Utils.PayloadToResult<${modelName}Payload>`
+  if (action === 'delete') return `$Utils.PayloadToResult<${modelName}Payload>`
+
+  assertNever(action, 'Unknown action: ' + action)
```

Comment:
I assume this chain of `if`s is due to `assertNever` not supporting `switch` statements?

Reply:
Not sure here, `assertNever` works with `switch`

Reply:
I think it's more likely because of stylistic preferences and not any specific reason. I'd probably write a `switch` but I don't mind either way, this looks fine to me too.

One could also write a ``Record<Exclude<Operation, `$${string}`>, () => string>`` or even ``Record<Exclude<Operation, `$${string}`>, string>`` and dispatch on it for the most concise syntax.